packages/hash/frontend/src/components/hooks/blockProtocolFunctions/knowledge/useBlockProtocolGetEntity.ts

```diff
@@ -0,0 +1,97 @@
+import { useLazyQuery } from "@apollo/client";
+import { useCallback } from "react";
+import { getEntityTypeRootedSubgraphQuery } from "../../../../graphql/queries/ontology/entity-type.queries";
+
+import {
+  GetEntityTypeRootedSubgraphQuery,
+  GetEntityTypeRootedSubgraphQueryVariables,
+  Query,
+  QueryKnowledgeEntityArgs,
+} from "../../../../graphql/apiTypes.gen";
+import { getKnowledgeEntity } from "../../../../graphql/queries/knowledge/entity.queries";
+import { GetEntityMessageCallback } from "./knowledge-shim";
+
+export const useBlockProtocolGetEntity = (): {
+  getEntity: GetEntityMessageCallback;
+} => {
+  const [getEntityTypeRootedSubgraphFn] = useLazyQuery<
+    GetEntityTypeRootedSubgraphQuery,
+    GetEntityTypeRootedSubgraphQueryVariables
+  >(getEntityTypeRootedSubgraphQuery, {
+    /** @todo reconsider caching. This is done for testing/demo purposes. */
```

Comment:
When do you want to address this, before merge?

Reply:
This is a copy and paste from the other BP methods, I don't think we should address before merging no.

Reply:
@thehabbos007 is there a task for this?

Reply:
Makes sense, thanks

Reply:
@thehabbos007 do we want an asana task to track this? Should it be part of the clean-up post merge?

Reply:
No task for this yet, i think it's good to have it be `no-cache` for a while, and we might only want to allow cached versions if the entity version IDs are passed in the query as well. I think caching could end up causing a lot of headache for frontend development, so best to leave it out for now.

We can create a task, though

Reply:
It's probably fine to leave without a task. We'll encounter this when we remove the shims and have actual BP methods